src/components/Home/Home.tsx

```diff
@@ -1,35 +1,66 @@
 import React, { useState, useEffect } from 'react'
+import { fetchGetResponse } from '../../utils/apiCalls'
+import { cleanMarketsData, cleanDetailsData } from '../../utils/utils'
 
 interface ApiMarkets {
   markets: {
     id: number,
+    distanceFromZip: number,
     marketName: string
-    //need logic to split the distance from the name to display the name
   }[],
   marketDetails: {
-    googlink: string,
-    address: string,
-    schedule: string,
-    products: string
-    // need logic for splitting products string into an array
+    street: string,
+    city: string,
+    state: string,
+    zip: string,
+    schedule: {
+      dayOfWeek: string,
+      time: string,
+      season: string
+    }[],
+    products: string[],
+    mapsLink: string
   }[]
 }
 
 const Home = ():JSX.Element => {
   const [ allMarkets, setMarkets ] = useState<ApiMarkets["markets"]>([])
   const [ marketDetails, setDetails ] = useState<ApiMarkets["marketDetails"]>([])
-  const [ error, setError ] = useState('')
-  
-  const getMarkets = async () => {
-    const url = 'http://search.ams.usda.gov/farmersmarkets/v1/data.svc/zipSearch?zip=01070'
-    setError('')
+  const [ error, setError ] = useState(0)
 
+  const getMarkets = async () => {
     try {
-      const response = await fetch(url)
-      const markets = await response.json()
-      setMarkets(markets)
-    } catch(error) {
-      setError(error.message)
+      let response = await fetchGetResponse('zipSearch?zip=01701');
+      checkForError(response)
+      let data = await response.json();
+      let cleanedData = cleanMarketsData(data.results)
+      setMarkets(cleanedData)
+      getDetails(cleanedData)
+    } catch (error) {
+      setError(error)
+    }
+  }
+
+  const getDetails = (filteredMarkets: ApiMarkets["markets"]) => {
+    Promise.all(filteredMarkets.map(currentMarket => {
+      return fetchGetResponse(`mktDetail?id=${currentMarket.id}`)
+      .then((response) => {
+        if (!response.ok) {
+          throw Error(response.statusText);
```

Comment:
Can we break this out into a checkForErrors function in utils? This is good for cleanliness of our Home component, as well as something I learned on the mid-mod if we throw Error in the same file as we try to render an Error component called Error, React throws a fit.

Reply:
Great suggestions! I have pushed commits to deal with these suggestions.